### ISSUE TITLE
Error (Xcode): Cycle inside Runner; building could produce unreliable results.

### ISSUE THREAD

**Issue by NM4ik** [flutter/flutter#142907]
### Type of Request

bug

### Infrastructure Environment

XCODE Version 15.2

### What is happening?

I add Notification Service Extension to the flutter project and immediately get an error when building on a physical device:

```
Cycle insideRunner; building could produce unreliable results.
Cycle details:
→ Target 'Runner': ExtractAppIntentsMetadata

After looking at other issues, I tried to change the build phases queue, but this led to another error: Command ValidateEmbeddedBinary failed with a nonzero exit code
``` 

<details><summary>flutter doctor -v</summary>
<p>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.9, on macOS 14.3 23D56 darwin-arm64, locale ru-RU)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.86.0)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for iPhone Виталя. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this
      Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources
``` 

</p>
</details> 

### Steps to reproduce

Step 1: Add target "Notification Service Extension" in xCode
Step 2: xCode -> Product -> Clean Build Folder
Step 3: xCode -> Build app (debug, release)

**Comment by Ivan981508** [comment]
I'm facing the same problem

**Comment by Josefpa** [comment]
![image](https://github.com/flutter/flutter/assets/67612575/470d7580-dfcc-4107-aded-ac7c15449ea9)

**Comment by dam-ease** [comment]
Hello @NM4ik.
Thanks for filing this. I'm able to reproduce this on my end as well.
Following the order as shared as a workaround in https://github.com/flutter/flutter/issues/135056#issuecomment-1735862765 works fine in my case, you can try this out as well.
This seems to be a known issue being tracked here:
https://github.com/flutter/flutter/issues/135056
https://forums.developer.apple.com/forums/thread/730974
Closing this as a duplicate of the existing issue.

**Comment by NM4ik** [comment]
Hello @dam-ease 
Thank you very much for your reply.

I created a text project and added a notification service extension to xCode. I changed the build phases there in the order of the queue and attach a screenshot.
<img width="1200" alt="image" src="https://github.com/flutter/flutter/assets/78036389/4930949e-66bd-49ed-9974-3df00007f622">

But I still get an error when building on a physical device. I have no idea at all how to solve it, because on my second computer on m2 pro, the same project runs without problems
